kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMConfigurationServiceImpl.java

```diff
@@ -295,6 +315,86 @@ private void decryptAndConvertPasswordProperties(Map<String, Object> modifiedPro
         }
     }
 
+    private void decryptAndConvertCertificatesProperties(Map<String, Object> modifiedProps, Set<String> interfaces) {
+
+        interfaces.forEach(interfaceName -> {
+            String key = String.format("net.interface.%s.config.802-1x.keystore.pid", interfaceName);
+            if (modifiedProps.containsKey(key)) {
+
+                Object prop = modifiedProps.get(key);
+
+                if (prop instanceof String) {
+                    String keystorePid = (String) prop;
+
+                    findAndDecodeCertificatesForInterface(interfaceName, modifiedProps,
+                            this.keystoreServices.get(keystorePid));
+                }
+            }
```

Comment:
You can leave the code as-is but consider using [Early Returns](https://medium.com/swlh/return-early-pattern-3d18a41bba8) in the future. I think the code is much easier to read:

```suggestion
            if (!modifiedProps.containsKey(key)) {
                return; // we cannot "continue" since we're inside a lambda
            }

            Object keystorePid = modifiedProps.get(key);
            if (!keystorePid instanceof String) {
                return;
            }

            findAndDecodeCertificatesForInterface(interfaceName, modifiedProps,
                    this.keystoreServices.get((String) keystorePid));
```